FutureNorthants/northants-website
# Frontend support for HeroImage structure

This adds support for the HeroImage structure component to the frontend. It depends on the following PRs:

- https://github.com/FutureNorthants/northants-design-system/pull/91
- https://github.com/FutureNorthants/northants-website/pull/214

## Setup

- Update the backend with the above PR, edit a service landing page and set a hero image
- Update the design-system with the above PR, then `yalc publish`
- Checkout this branch on the frontend, then `yalc add northants-design-system` to pull in the new design system work and run `yarn`

## Testing
- Run `yarn build` and verify no errors
- Run `yarn dev`
- In a new terminal window run `yarn codegen` and verify no changes are made
- Visit the page you edited and verify that the hero image is displayed at the top under the normal header banner area, no breadcrumb is shown, nor any note about the website being new
- Go back to the CMS and remove the hero image from the page and reload it in the frontend. Verify it returns to normal appearance.

## Deployment
Once all PRs are approved, the design system needs to be pushed to npm as a new version and then this branch's `package.json` and `yarn.lock` updated to use the latest version of the design system. Deploy backend first then frontend.
